fluent/client/client.go

```diff
@@ -43,12 +43,18 @@ type Client struct {
 	Hostname string
 }
 
+// ServerAddress holds details about URLs of the server to connect
 type ServerAddress struct {
-	Hostname string
-	Port     int
+	Hostname string // Hostname of the server to connect
+	Port     int    // Port of the server
+	URLPath  string // Optional Path to connect on server, it should start with trailing slash e.g. /events/ingest
 }
 
-func (sa ServerAddress) String() string {
+func (sa *ServerAddress) String() string {
+	if len(sa.URLPath) != 0 {
```

Comment:
Port isn't required for websockets. Update comment to clarify. Then do something like:

```go
func (sa ServerAddress) String() string {
        hn :=  sa.Hostname

	if sa.Port != 0 {
	      hn = fmt.Sprintf("%s:%d", hn, sa.Port)
	}

	return path.Join(hn, sa.Path)
}
```

 Note that `String()` doesn't need a pointer receiver

Reply:
Updated